mne/preprocessing/faster.py

```diff
@@ -0,0 +1,463 @@
+    epochs : Instance of Epochs
+        The epochs for which bad channels need to be marked
+    picks : list of int | None
+        Channels to operate on. Defaults to EEG channels.
+    thres : float
+        The threshold value, in standard deviations, to apply. A channel
+        crossing this threshold value is marked as bad. Defaults to 3.
+    use_metrics : list of str
+        List of metrics to use. Can be any combination of:
+            'variance', 'correlation', 'hurst', 'kurtosis', 'line_noise'
+        Defaults to all of them.
+    max_iter : int
+        The maximum number of iterations performed during outlier detection
+        (defaults to 2, original FASTER paper uses 1).
+    return_by_metric : bool
+        Whether to return the bad channels as a flat list (False, default) or
+        as a dictionary with the names of the used metrics as keys and the
+        bad channels found by this metric as values.
+
+    Returns
+    -------
+    bads : list of str
+        The names of the bad EEG channels.
+    """
+    metrics = {
+        'variance': lambda x: np.var(x, axis=1),
+        'correlation': lambda x: np.mean(
+            np.ma.masked_array(np.corrcoef(x),
+                               np.identity(len(x), dtype=bool)), axis=0),
+        'hurst': lambda x: _hurst(x),
+        'kurtosis': lambda x: kurtosis(x, axis=1),
+        'line_noise': lambda x: _freqs_power(x, epochs.info['sfreq'],
+                                             [50, 60]),
+    }
+
+    if picks is None:
+        picks = pick_types(epochs.info, meg=True, eeg=True, exclude=[])
+    if use_metrics is None:
+        use_metrics = metrics.keys()
+
+    # Concatenate epochs in time
+    data = epochs.get_data()
+    data = data.transpose(1, 0, 2).reshape(data.shape[1], -1)
+    data = data[picks]
+
+    # Find bad channels
+    bads = defaultdict(list)
+    for ch_type, chs in _by_ch_type(epochs.info, picks):
+        logger.info('Bad channel detection on %s channels:' % ch_type.upper())
+        for m in use_metrics:
+            s = metrics[m](data[chs])
+            b = [epochs.ch_names[picks[chs[i]]]
+                 for i in find_outliers(s, thres, max_iter)]
+            logger.info('\tBad by %s: %s' % (m, b))
+            bads[m].append(b)
+
+    bads = dict([(k, np.concatenate(v).tolist()) for k, v in bads.items()])
+    if return_by_metric:
+        return bads
+    else:
+        return np.unique(np.concatenate(list(bads.values()))).tolist()
+
+
+def faster_bad_epochs(epochs, picks=None, thres=3, use_metrics=None,
+                      max_iter=2, return_by_metric=False):
+    """Implements the second step of the FASTER algorithm.
+
+    This function attempts to automatically mark bad epochs by performing
+    outlier detection.
+
+    Parameters
+    ----------
+    epochs : Instance of Epochs
+        The epochs to analyze.
+    picks : list of int | None
+        Channels to operate on. Defaults to EEG channels.
+    thres : float
```

Comment:
I mean a single function that calls all the FASTER steps in sequence. I've described it elsewhere. The arguments to the functions look very similar to me. No point in asking the user to write 5 lines when it can be done in one line.